### DISCUSSION TITLE
s3 connection type not found

### DISCUSSION THREAD

**Discussion by VidhyaSree-N** [apache/airflow#27129] [Q&A]
I have installed necessary providers for s3 connection but in airflow ui I cannot see s3 connection type in drop down.
Any kind of help is appreciated.
Thank you.

Providers instaled:
pip install apache-airflow[s3]
pip install apache-airflow-providers-amazon

**Comment by Taragolis** [comment]
This connection type was removed since version 6.0.0 of provider

See the changelog: 
- https://airflow.apache.org/docs/apache-airflow-providers-amazon/6.0.0/index.html#breaking-changes
- https://pypi.org/project/apache-airflow-providers-amazon/

**Reply by VidhyaSree-N** [reply]
Okay Thank you.
How do i connect to s3 with latest versions?

**Reply by Taragolis** [reply]
Actually nothing changes since all other version of amazon provider. The main connection is AWS Connection and S3 was an alias.

```text
In this version of provider Amazon S3 Connection (conn_type="s3") removed 
due to the fact that it was always an alias to AWS connection conn_type="aws"
In practice the only impact is you won’t be able to test the connection in the web UI / API.
In order to restore ability to test connection you need to change connection type from Amazon S3 (conn_type="s3") to Amazon Web Services (conn_type="aws") manually.
```

AWS Connection, see: https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/connections/aws.html

**Reply by VidhyaSree-N** [reply]
When i tried to connect via aws it gave me this error
An error occurred (InvalidClientTokenId) when calling the GetCallerIdentity operation: The security token included in the request is invalid.
Access token and secret key are valid.I did test them locally.

Any kind of help is appreciated.

Thank you

**Reply by Taragolis** [reply]
This error usual refers to missing `aws_session_token` so you need provide not only `aws_access_key_id`  and `aws_secret_access_key` in your case. 

However I want to warn you that if you required `aws_session_token` that mean you use temporary credentials which expired in 15 min to 48 hours (depend on settings for role) and you need manually change you credentials manually in connection. Try to switch to use permanent credentials or if you run Airflow in AWS cloud you might consider to use EC2 Instance Profile, ECS Execution Task role and etc (depend on where your deployment)

Anyway this mostly refers to AWS Credentials not to Airflow, I found some useful links for you (this not all completed list):
- https://docs.aws.amazon.com/sdkref/latest/guide/feature-static-credentials.html
- https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_use-resources.html
- https://docs.aws.amazon.com/IAM/latest/UserGuide/id.html
- https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles.html
- https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-iam-roles.html

**Reply by VidhyaSree-N** [reply]
All the documents mentioned,by default points aws connection but we have a custom s3 local connection that doesn’t start with url aws://

Any thoughts on how to connect to custom s3 connection with different url and no region specified.

Thank you

**Reply by Taragolis** [reply]
Hey @VidhyaSree-N  `aws://` it is internals of Airflow and not related to any of [S3 URI](https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-bucket-intro.html)

I would recommend check different examples included into the repo, e.g.:
https://github.com/apache/airflow/blob/main/tests/system/providers/amazon/aws/example_s3.py

**Reply by Wesseldr** [reply]
@VidhyaSree-N Did you manage to get it working ? If you have please add the answer. I'm hitting over a day to get it connecting to MinIO S3

Thank You!

**Reply by Taragolis** [reply]
@Wesseldr I would recommend [open new discussion](https://github.com/apache/airflow/discussions/new?category=q-a) with proper description of you problem. More detail about what you already tried, which sources you use and what problem you have might help to get right answer quicker. 

And a bit more, if you found the right answer you could extend Airflow documentation by your contribution, Apache Airflow is OSS which in not only Open Source but also Open for Contribution so you might also solve problem other Airflow users.

Please also note, Amazon Provider intend to use with a native AWS environment rather than OSS implementations and it is possible that some part would not work in MinIO properly due to lack of implementation some specific AWS API.

**Reply by Taragolis** [reply]
And also I would recommend to check what you provide to [Amazon Web Services Connection ](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/connections/aws.html) because I see some kind of articles in the Internet about Airflow + Minio which suggest to use something which not never beens supported by Amazon Provider maybe it is a root of your (and others) problem

**Reply by Wesseldr** [reply]
@Taragolis Thank you. I got it working, I've found another post of you that helped me out. 
https://github.com/apache/airflow/discussions/26979#discussioncomment-3928659

I was tricked by the test-connection. 
"Test_connection is failing, it doesn't necessarily mean that the connection won't work."

**Comment by sksirajbasha** [comment]
@Taragolis what we are trying to achieve is creating a connection to our orgs storage which is similar to S3 and uses an extra in connection like this (in version where S3 - connection type was available)

  {
    "endpoint_url":"https://customobjectstorage.mycompany.com",
    "service_name":"s3",
    "aws_access_key_id":"accesskey",
    "aws_secret_access_key":"secretkey"
    }

what I see in the python file above is to create and use local S3 buckets.

We intend to present a TOSS bucket as DAGs storage mount to our airflow container which does not have NFS mount options because of our private cloud restrictions.

We will use a similar bucket for remote logging too and both of these need a working airflow connection.

This was done in my org and works when the legacy(s3) was available in older versions of airflow.

If we try similar configs now, the URL endpoint fails ans also asks for a region.

**Reply by Taragolis** [reply]
> This was done in my org and works when the legacy(s3) was available in older versions of airflow.

Obtain credentials independent of connection type and depend only on AwsBaseHook / AwsGenericHook.
AWS S3 Connection was only UI.

So if previously something work and now it broken unlikely this happen because one connection removed from the UI.
More possible that the actual reason changes in `boto3` / `botocore` internals especially if you use some OSS implementation of S3

> "service_name":"s3",

This extra argument doesn't use. Please consider to check available parameters https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/connections/aws.html#configuring-the-connection
